src/main/java/jewellery/inventory/service/OrganizationService.java

```diff
@@ -67,6 +80,37 @@ private List<Organization> getAll() {
     return organizationRepository.findAll();
   }
 
+  private void validateOrganizationProductsAndResources(Organization organization) {
+    if (!organization.getProductsOwned().isEmpty()) {
+      throw new OrganizationProductsException(organization.getId());
+    }
+    if (!organization.getResourceInOrganization().isEmpty()) {
+      throw new OrganizationResourcesException(organization.getId());
+    }
+  }
+
+  public void validateCurrentUserPermission(
```

Comment:
When having validation methods, it is good practice to return type (boolean for example). Probably better to return boolean and if it doesn't throw exception to return true.
Or the other option is to return false if it is not valid and to throw the exception from where validation is called.

Reply:
I don't quite agree. We are using the validate methods to throw exception if the business logic doesn't allow us to continue with the given operation. This sounds reasonable to me as we are interrupting the current logic from continuing and are directly creating a response out of this failed validation. What you're suggesting will do the same but will add more complexity to the code due to additional if statements. The only benefit I see is if we want to include multiple error messages to the response. In this ticket it could make sense if the organization has both products and resources available when trying to delete it. This is an actual case here but I don't think it's worth it to explore it deeper for now.